src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double NEW_ARRAY_CAPACITY_FACTOR = 1.5;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        rangeCheckForAdd(index);
+        checkCapacity();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            checkCapacity();
+            elementData[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheckForAdd(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheckForAdd(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T oldElementData;
+        try {
+            oldElementData = (T) elementData[index];
+            rangeCheckForAdd(index);
+            System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new ArrayListIndexOutOfBoundsException(index + " is invalid index");
+        }
+        --size;
+        return oldElementData;
     }
 
     @Override
     public T remove(T element) {
```

Comment:
The method is overloaded, try to do it simpler
find the index of the element and call the method `remove(int index)`